firecrest/v2/_async/Client.py

```diff
@@ -826,84 +1012,181 @@ async def rm(
     async def upload(
         self,
         system_name: str,
-        local_file: str | pathlib.Path | BytesIO,
+        local_file: str | pathlib.Path,
         directory: str,
         filename: str,
```

Comment:
I was thinking about merging `directory` and `filename` into a single argument, but I'm not sure if that could get confusing since the argument could be interpreted as just a target directory without changing the name or a directory with a new filename.

Otherwise, would it make sense to rename directory to something like `target_path` or similar like in the `download`?